src/components/Modal/index.tsx

```diff
@@ -0,0 +1,84 @@
+// reusable modal component
+
+import { Dispatch, Fragment, ReactNode, SetStateAction } from "react";
+import { Dialog, Transition } from "@headlessui/react";
+import { BsXCircle } from "react-icons/bs";
+
+type ModalProps = {
+  showModal: boolean;
+  setShowPopup: Dispatch<SetStateAction<boolean>>;
+  title?: string;
+  width?: string;
+  height?: string;
+  children?: ReactNode;
+};
+
+const Modal = ({
+  showModal,
+  setShowPopup,
+  title,
+  width = "w-full sm:max-w-lg md:max-w-2xl",
+  height = "h-full",
+  children
+}: ModalProps) => {
+  return (
+    <>
+      <Transition appear show={showModal} as={Fragment}>
+        <Dialog
+          as="div"
+          className="relative z-10 "
```

Comment:
Remove this trailing space

Reply:
Thank you for taking the time to review my code. Your feedback is valuable to me, and I really appreciate your efforts to help me improve！